File: two-pointers/11-max_water.py
class Solution(object):
    def maxArea(self, height):
        l,r=0,len(height)-1
        max_water=0
        while l<r:
            width=r-l 
            area=min(height[l], height [r])*width
            max_water=max(area,max_water)
            if height[l]<height[r]:
                l+=1
            else:
                r-=1
        return max_water 

File: two-pointers/11-max-water.c
int maxArea(int* height, int heightSize) {
    int l = 0;
    int r = heightSize - 1;
    int max_water = 0;
    
    while (l < r) {
        int h = height[l] < height[r] ? height[l] : height[r];
        int water = h * (r - l);
        
        if (water > max_water) {
            max_water = water;
        }
        
        if (height[l] < height[r]) {
            l++;
        } else {
            r--;
        }
    }
   
